indico/modules/events/contributions/models/contributions.py

```diff
@@ -547,6 +547,14 @@ def can_manage(self, user, permission=None, allow_admin=True, check_parent=True,
                                         explicit_permission=explicit_permission) and
                 session_coordinator_priv_enabled(self.event, 'manage-contributions')):
             return True
+        # Submitters can manage their own contributions if configured
+        from indico.modules.events.abstracts.settings import abstracts_settings
+        submitters_can_manage = abstracts_settings.get(self.event, 'allow_submitters_edit_contributions')
+        if submitters_can_manage and any(
+            principal.has_management_permission('submit') and principal.user == user
+            for principal in self.acl_entries
+        ):
+            return True
```

Comment:
The check could be simplified a lot by passing `permission='submit'` to the super `can_manage()` check at the beginning when submitters are allowed to edit.

However, they should not have full management permissions on the contribution - this would also allow things such as deleting it or even changing its schedule. They should just be able to edit basic data like title, description, speakers, etc.